File: software/examples/java/ExamplePixelMatrix.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletOLED128x64V2;

public class ExamplePixelMatrix {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your OLED 128x64 Bricklet
	private static final String UID = "XYZ";
	private static final short SCREEN_WIDTH = 128;
	private static final short SCREEN_HEIGHT = 64;

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletOLED128x64V2 oled = new BrickletOLED128x64V2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Clear display
		oled.clearDisplay();

		// Draw checkerboard pattern
		boolean[] pixels = new boolean[SCREEN_HEIGHT * SCREEN_WIDTH];

		for (short h = 0; h < SCREEN_HEIGHT; h++) {
			for (short w = 0; w < SCREEN_WIDTH; w++) {
				pixels[h*SCREEN_WIDTH + w] = (h / 8) % 2 == (w / 8) % 2;
			}
		}

		oled.writePixels(0, 0, SCREEN_WIDTH-1, SCREEN_HEIGHT-1, pixels);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/vbnet/ExamplePixelMatrix.vb
Imports System
Imports System.Math
Imports Tinkerforge

Module ExamplePixelMatrix
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your OLED 128x64 Bricklet 2.0
    Const WIDTH As Integer = 128
    Const HEIGHT As Integer = 64

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim oled As New BrickletOLED128x64V2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Clear display
        oled.ClearDisplay()

        ' Draw checkerboard pattern
        Dim pixels() As Boolean = New Boolean(HEIGHT*WIDTH) {}

        For row As Integer = 0 To HEIGHT - 1
            For column As Integer = 0 To WIDTH - 1
                pixels(row*WIDTH + column) = (row \ 8) Mod 2 = (column \ 8) Mod 2
            Next column
        Next row

		oled.WritePixels(0, 0, WIDTH-1, HEIGHT-1, pixels)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
